internal/scaffold/helm/watches.go

```diff
@@ -43,6 +53,56 @@ func (s *WatchesYAML) GetInput() (input.Input, error) {
 	return s.Input, nil
 }
 
+// UpdateHelmWatchForResource checks for duplicate GVK, and appends to existing Watch.yaml file.
+func UpdateHelmWatchForResource(r *scaffold.Resource, absProjectPath string) error {
+	watchFilePath := filepath.Join(absProjectPath, WatchesYamlFile)
+	watchYAML, err := ioutil.ReadFile(watchFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read watch manifest %v: %v", watchFilePath, err)
+	}
+	if len(watchYAML) == 0 {
+		return fmt.Errorf("empty Watch File at: %v", absProjectPath)
+	}
+
+	watchList := []watches.Watch{}
+	err = yaml.Unmarshal(watchYAML, &watchList)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal watch config %v ", err)
+	}
+
+	gvk := schema.GroupVersionKind{
+		Version: r.Version,
+		Group:   r.FullGroup,
+		Kind:    r.Kind,
+	}
+
+	for _, watch := range watchList {
+		if watch.GroupVersionKind == gvk {
+			// dupe detected
+			return fmt.Errorf("duplicate GVK: %v", watch.GroupVersionKind.String())
+		}
+	}
+	newWatch := watches.Watch{
+		GroupVersionKind: gvk,
+		ChartDir:         HelmChartsDir + "/" + r.LowerKind,
```

Comment:
To fix the issue @camilamacedo86 noted about the mismatched chart directory name, we need to pass `chart` to this function and use that to set `ChartDir`

Reply:
@joelanford I am guessing this is  an existing bug as well if I understand 
Currently, when creating new watch, as you see below `s.Resource.LowerKind` is being set.
```go
type WatchesYAML struct {
	input.Input

	Resource      *scaffold.Resource
	HelmChartsDir string
	ChartName     string
}

// GetInput gets the scaffold execution input
func (s *WatchesYAML) GetInput() (input.Input, error) {
	if s.Path == "" {
		s.Path = WatchesYamlFile
	}
	s.HelmChartsDir = HelmChartsDir
	s.TemplateBody = watchesYAMLTmpl
	if s.ChartName == "" {
		s.ChartName = s.Resource.LowerKind
	}
	return s.Input, nil
}
```

Reply:
That's only if ChartName is empty. We set it [here](https://github.com/operator-framework/operator-sdk/blob/2322d7811d0c46ed8d2a6cd9a50e3331a2b109e6/cmd/operator-sdk/new/cmd.go#L403).